lib/src/io.cpp

```diff
@@ -405,23 +405,44 @@ record& extract(stream& file, long long tell, long long bytes, record& rec) noex
     }
 }
 
-stream_offsets findoffsets( dl::stream& file) noexcept (false) {
+stream_offsets findoffsets( dl::stream& file, dl::error_handler& errorhandler)
+noexcept (false) {
     stream_offsets ofs;
 
-    std::int64_t offset = 0;
+    std::int64_t lr_offset = 0;
+    std::int64_t lrs_offset = 0;
+
     int has_successor = 0;
     char buffer[ DLIS_LRSH_SIZE ];
 
+    const auto handle = [&]( const std::string& problem ) {
+        const auto context = "load (findoffsets)";
+        errorhandler.log(dl::error_severity::CRITICAL, context, problem, "",
+                         "Stopped file processing");
+        ofs.broken.push_back( lr_offset );
+    };
+
     int len = 0;
     while (true) {
-        file.seek(offset);
-        file.read(buffer, DLIS_LRSH_SIZE);
+        file.seek(lrs_offset);
+        try {
+            file.read(buffer, DLIS_LRSH_SIZE);
+        } catch (std::exception& e) {
+            handle(e.what());
+            break;
+        }
+
         if (file.eof()) {
-            check_for_truncation(file, offset);
+            if (is_truncated(file, lrs_offset)) {
+                const auto problem = "findoffsets: file truncated";
```

Comment:
But probably not much. Have you measured this? Im not so sure it will have a noticeable impact. By reading the last byte of each segment what we essentially do is read _5 successive bytes_ instead of 4. But it would be good to measure anyway

Reply:
As per our investigation just combined seeks.

Reply:
Message should be `file truncated in Logical Record`, otherwise all good